cmd/controller/app/controller.go

```diff
@@ -230,18 +230,29 @@ func buildControllerContext(ctx context.Context, opts *options.ControllerOptions
 		return nil, nil, fmt.Errorf("error creating kubernetes client: %s", err.Error())
 	}
 
+	// check if the gateway API CRDs are available
+	var gatewayAvailable bool
+	d := cl.Discovery()
+	resources, err := d.ServerResourcesForGroupVersion(gwapi.GroupVersion.String())
+	switch {
+	case apierrors.IsNotFound(err):
+		gatewayAvailable = false
+		log.Info("the Gateway API CRDs do not seem to be present, cert-manager will keep retrying watching for them")
```

Comment:
When I start cert-manager without the CRDs installed, is it going to still print the messages about the reflector not being able to list *v1alpha1.Gateway even though cert-manager would require a restart for `gatewayAvailable` to be set to true?